tests/Api/Shop/AddressesPostTest.php

```diff
@@ -0,0 +1,102 @@
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Tests\Api\Shop;
+
+use Sylius\Component\Addressing\Model\CountryInterface;
+use Sylius\Component\Core\Model\CustomerInterface;
+use Sylius\Tests\Api\JsonApiTestCase;
+use Symfony\Component\HttpFoundation\Response;
+
+final class AddressesPostTest extends JsonApiTestCase
+{
+    /** @test */
+    public function it_denies_access_to_a_create_an_address_for_not_authenticated_user(): void
+    {
+        $fixtures = $this->loadFixturesFromFiles(['authentication/customer.yaml', 'country.yaml']);
+        /** @var CustomerInterface $customer */
+        $customer = $fixtures['customer_Oliver'];
+        /** @var CountryInterface $country */
+        $country = $fixtures['country_US'];
+
+        $bodyRequest = $this->createBodyRequest($customer, $country);
+
+        $this->client->request(
+            'POST',
+            '/api/v2/shop/addresses',
+            [],
+            [],
+            ['CONTENT_TYPE' => 'application/ld+json', 'HTTP_ACCEPT' => 'application/json'],
+            json_encode($bodyRequest)
+        );
+
+        $response = $this->client->getResponse();
+        $this->assertSame(Response::HTTP_INTERNAL_SERVER_ERROR, $response->getStatusCode());
+    }
+
+    /** @test */
+    public function it_creates_new_address_for_logged_customer(): void
+    {
+        $fixtures = $this->loadFixturesFromFiles(['authentication/customer.yaml', 'country.yaml']);
+        /** @var CustomerInterface $customer */
+        $customer = $fixtures['customer_Oliver'];
+        /** @var CountryInterface $country */
+        $country = $fixtures['country_US'];
+
+        $this->client->request(
+            'POST',
+            '/api/v2/shop/authentication-token',
+            [],
+            [],
+            ['CONTENT_TYPE' => 'application/json', 'HTTP_ACCEPT' => 'application/json'],
+            json_encode(['email' => $customer->getEmailCanonical(), 'password' => 'sylius'])
+        );
+
+        $token = json_decode($this->client->getResponse()->getContent(), true)['token'];
+        $authorizationHeader = self::$container->getParameter('sylius.api.authorization_header');
+
+        $header['HTTP_' . $authorizationHeader] = 'Bearer ' . $token;
+
+        $bodyRequest = $this->createBodyRequest($customer, $country);
+
+        $this->client->request(
+            'POST',
+            '/api/v2/shop/addresses',
+            [],
+            [],
+            array_merge($header, ['CONTENT_TYPE' => 'application/ld+json', 'HTTP_ACCEPT' => 'application/json']),
+            json_encode($bodyRequest)
+        );
+
+        $response = $this->client->getResponse();
+
+        $this->assertResponse($response, 'shop/post_addresses_response', Response::HTTP_CREATED);
+    }
+
+    private function createBodyRequest(CustomerInterface $customer, CountryInterface $country): array
+    {
+        return [
+            'customer' => '/api/v2/shop/customers/'.$customer->getId(),
```

Comment:
only logged customers can create addresses so IMO this filed in the body isn't necessary (it should be added in backend automatically) but maybe it should be different task:)

but:
```suggestion
            'customer' => '/api/v2/shop/customers/' . $customer->getId(),
```